### ISSUE TITLE
Add `seekable` method for `mmap.mmap`

### ISSUE THREAD

**Issue by technillogue** [python/cpython#111835]
# Bug report

### Bug description:

`mmap.mmap` has a `seek()` method, but no `seekable()` method. Because of this, it cannot be used an argument for ZipFile, which requires a file-like object with seekable. A small patch is enough to fix this

```python
class SeekableMmap(mmap.mmap):
    def seekable(self):
        return True
```

Now SeekableMmap can be passed to ZipFile without issue. It's easy to fix, but I can't think of any reasons why mmap shouldn't have a seekable method.

### CPython versions tested on:

3.8, 3.11

### Operating systems tested on:

Linux

### Linked PRs
* gh-111852
* gh-111865

**Comment by serhiy-storchaka** [comment]
`mmap.seek()` returns None. The `seek()` method of other file-like objects return the new file position. Would not it cause issues?

**Comment by technillogue** [comment]
I also opened a PR https://github.com/python/cpython/pull/111865 for this, I just took a while between pushing the code and opening the PR.

**Comment by corona10** [comment]
> mmap.seek() returns None. The seek() method of other file-like objects return the new file position. Would not it cause issues?

Yeah, in that case, we can make a seek() method to return offset position. It will break the user's code actually?
In my naive idea, people cannot do something with None value.

**Comment by corona10** [comment]
@technillogue 

https://github.com/python/cpython/pull/111852 is landed, this feature will be available from CPython 3.13
Thanks for the suggestion!

@serhiy-storchaka @vstinner 
Thanks for your valuable feedback and reviews!
